src/gdb_main.c

```diff
@@ -707,7 +661,58 @@ static void handle_z_packet(char *packet, const size_t plen)
 		gdb_putpacketz("OK");
 }
 
-void gdb_main(void)
+void gdb_main(char *pbuf, size_t pbuf_size, size_t size)
+{
+	gdb_main_loop(&gdb_controller, pbuf, pbuf_size, size, false);
+}
+
+/* halt target */
+void gdb_halt_target(void)
+{
+	if (cur_target)
+		target_halt_request(cur_target);
+	else
+		/* Report "target exited" if no target */
+		gdb_putpacketz("W00");
+}
+
+/* poll running target */
+void gdb_poll_target(void)
 {
-	gdb_main_loop(&gdb_controller, false);
+	target_addr_t watch;
```

Comment:
Please don't use C89 style declarations - having the if statement first, then declaring watch and declare-initialising reason helps the compiler and removes a bug possibility.